bookkeeper-server/src/main/java/org/apache/bookkeeper/client/ReadOnlyLedgerHandle.java

```diff
@@ -326,11 +326,15 @@ CompletableFuture<Versioned<LedgerMetadata>> closeRecovered() {
                     return builder.withClosedState().withLastEntryId(lac).withLength(len).build();
                 },
                 this::setLedgerMetadata).run();
-        f.thenRun(() -> {
-                synchronized (metadataLock) {
-                    newEnsemblesFromRecovery.clear();
-                }
-            });
+        f.whenComplete((result, exception) -> {
+            synchronized (metadataLock) {
```

Comment:
There is one thing I care about:
Shall we clear `newEnsemblesFromRecovery` when the metadata update fails? 
Line_314 support `newEnsemblesFromRecovery` to do some work. If we clear it, is there some influence for it?

Reply:
@horizonzy it's ok, 
I don't think the problem you mentioned has anything to do with this pr?
I understand this logic is
when finished to `closeRecovered`, clean up data for `newEnsemblesFromRecovery` at the end

Reply:
Fine.